src/OrleansAzureUtils/MultiClusterNetwork/GossipTableInstanceManager.cs

```diff
@@ -0,0 +1,327 @@
+        }
+
+
+        internal async Task<bool> TryUpdateGatewayEntryAsync(GatewayEntry entry, GossipTableEntry row, string eTag)
+        {
+            //Debug.Assert(row.ETag == eTag);
+            //Debug.Assert(row.PartitionKey == GlobalServiceId);
+            //Debug.Assert(row.RowKey == GossipTableEntry.ConstructRowKey(entry.SiloAddress, entry.ClusterId));
+            
+            row.Status = entry.Status.ToString();
+            row.GossipTimestamp = entry.HeartbeatTimestamp;
+
+            return (await TryUpdateTableEntryAsync(row, eTag).ConfigureAwait(false) != null);
+        }
+
+        internal Task<bool> TryDeleteGatewayEntryAsync(GossipTableEntry row, string eTag)
+        {
+            return TryDeleteTableEntryAsync(row, eTag);
+        }
+
+        internal async Task<int> DeleteTableEntries()
+        {
+            var entries = await storage.ReadAllTableEntriesForPartitionAsync(GlobalServiceId).ConfigureAwait(false);
+            var entriesList = new List<Tuple<GossipTableEntry, string>>(entries);
+            if (entriesList.Count <= AzureTableDefaultPolicies.MAX_BULK_UPDATE_ROWS)
+            {
+                await storage.DeleteTableEntriesAsync(entriesList).ConfigureAwait(false);
+            }
+            else
+            {
+                List<Task> tasks = new List<Task>();
+                foreach (var batch in entriesList.BatchIEnumerable(AzureTableDefaultPolicies.MAX_BULK_UPDATE_ROWS))
+                {
+                    tasks.Add(storage.DeleteTableEntriesAsync(batch));
+                }
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
+            return entriesList.Count;
+        }
+
+
+        /// <summary>
+        /// Try once to conditionally update a data entry in the Azure table. Returns null if etag does not match.
+        /// </summary>
+        private async Task<string> TryUpdateTableEntryAsync(GossipTableEntry data, string dataEtag, [CallerMemberName]string operation = null)
+        {
+            try
+            {
+                return await storage.UpdateTableEntryAsync(data, dataEtag).ConfigureAwait(false);
+            }
+            catch (Exception exc)
+            {
+                HttpStatusCode httpStatusCode;
+                string restStatus;
+                if (!AzureStorageUtils.EvaluateException(exc, out httpStatusCode, out restStatus)) throw;
+
+                if (logger.IsVerbose2) logger.Verbose2("{0} failed with httpStatusCode={1}, restStatus={2}", operation, httpStatusCode, restStatus);
+                if (AzureStorageUtils.IsContentionError(httpStatusCode)) return null;
+
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Try once to insert a new data entry in the Azure table. Returns null if etag does not match.
+        /// </summary>
+        private async Task<string> TryCreateTableEntryAsync(GossipTableEntry data, [CallerMemberName]string operation = null)
```

Comment:
Same here - the returned value is only ever checked for null, so it's used as a Boolean value.